Sources/MongoSwift/Collection.swift

```diff
@@ -25,16 +25,23 @@ public struct AggregateOptions: BsonEncodable {
     /// The index to use for the aggregation. The hint does not apply to $lookup and $graphLookup stages.
     // let hint: Optional<(String | Document)>
 
+    /// A ReadConcern to use for this operation. 
+    let readConcern: ReadConcern?
```

Comment:
could we put `readConcern` and `writeConcern` in a base class instead of explicitly defining it for every options type? `ReadConcernable` 😄

Reply:
Hmm, I guess it makes it hard to add it to the field-wise initializer :thinking: But it might make it so you don't need `skipKeys` anymore - you could ignore fields brought in by known decorators, leaving just the base struct as the codable type for serialization. Worth a quick check I guess?

Reply:
we'd then have to convert every options type to a `class`, `struct`s have no inheritance 😢 
I tried to think about how to do it with a `protocol` + an `extension` of that `protocol` as well, but you can't define stored properties in protocol extensions (which is what we would want for the `readConcern` property), only computed properties and methods.